### PR TITLE
Fix up some type signatures...

### LINKED ISSUE #5901: Tag values use incorrect fully qualified names.
The `tag_values` field provided within suggestion database should always contain fully qualified names, so they can be resolved on IDE side and used as suggestions in dropdowns.

Currently for some types, the provided names within `tag_values` are either not fully qualified (but work with the right import), or just completely incorrect.

### Example with non-FQN values
`Text.trim` method, `where` argument.
image
Provided argument data in suggestion database:
```json
{
  "name": "where",
  "reprType": "Location.Start | Location.End | Location.Both",
  "isSuspended": false,
  "hasDefault": true,
  "defaultValue": null,
  "tagValues": [
    "Location.Start",
    "Location.End",
    "Location.Both"
  ]
}
```
Provided values are short. This happen to work in IDE, since the right import is usually already present. Unfortunately it means we cannot add that import in case it is necessary.

Expected tagValues:
```json
  "tagValues": [
    "Standard.Base.Data.Text.Location.Start",
    "Standard.Base.Data.Text.Location.End",
    "Standard.Base.Data.Text.Location.Both"
  ]
```

### Example with invalid value
`File.read` method, `on_problems` argument.
image
Provided argument data in suggestion database:
```json
{
  "name": "on_problems",
  "reprType": "Standard.Base.Error.Problem_Behavior.Problem_Behavior",
  "isSuspended": false,
  "hasDefault": true,
  "defaultValue": null,
  "tagValues": [
    "Standard.Base.Error.Problem_Behavior.Ignore",
    "Standard.Base.Error.Problem_Behavior.Report_Warning",
    "Standard.Base.Error.Problem_Behavior.Report_Error"
  ]
}
```

The provided tag values contain invalid FQNs. The module and type names has been incorrectly deduplicated. The correct values would be:
```json
  "tagValues": [
    "Standard.Base.Error.Problem_Behavior.Problem_Behavior.Ignore",
    "Standard.Base.Error.Problem_Behavior.Problem_Behavior.Report_Warning",
    "Standard.Base.Error.Problem_Behavior.Problem_Behavior.Report_Error"
  ]
```

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

Align any type signatures with a mismatch in count between types and arguments.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Geo/0.0.0-dev/src/Geo_Json.enso [hunk 1]
```diff
@@ -6,7 +6,7 @@ import Standard.Table.Errors.Invalid_JSON_Format
 import project.Helpers
 import project.Helpers.Object_Type
 
-geo_json_to_table : (Vector | JS_Object) -> Table
+geo_json_to_table : (Vector | JS_Object) -> (Vector Text) -> Table
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Geo/0.0.0-dev/src/Geo_Json.enso]: Don't we generally leave the brackets off?

File: distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Base_Generator.enso [hunk 2]
```diff
@@ -138,7 +138,7 @@ make_function name =
 
    Arguments:
    - sql_code: The SQL code for the constant.
-make_constant : Text -> Vector Builder -> Builder
+make_constant : Text -> (Vector Builder -> Builder)
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Base_Generator.enso:141]: Is (->) left-associative, or is this for clarity?

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Base_Generator.enso (hunk 2 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Base_Generator.enso:141]: this function returns a function which takes a `Vector Builder` and returns a `Builder`.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Runtime.enso [hunk 3]
```diff
@@ -78,7 +78,7 @@ no_inline ~action = @Builtin_Method "Runtime.no_inline"
      Print something to the console without it being inlined.
 
          Runtime.no_inline_with_arg IO.println "Hi!"
-no_inline_with_arg : Any -> Any
+no_inline_with_arg : Any -> Any -> Any
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Runtime.enso]: shouldn't this be
```suggestion
no_inline_with_arg : (Any -> Any) -> Any -> Any
```
?

**Review by radeusgd — APPROVED** [review]

**Pull request merged**